### DISCUSSION TITLE
Unable to call book POST, receiving 400 error

### DISCUSSION THREAD

**Discussion by teddywagner** [Alkaar/resy-booking-bot#108] [Q&A] [answered]
I have been reading through this repository and its resources for the last couple of weeks and would like to say that this is a very cool community and I have so much respect for the project and creator.

I've been working through the Api myself as a sort of side-project and have hit a wall in regard to the /book request.

I'm sure I'm using the right parameters for this with book_token and struct_payment_id, but the error I'm receiving is
```
"book_token": "An invalid book token was provided."
```
I'm extracting the token from /details via `book_token > value` so I'm unsure why the token is invalid.

Lastly, I've been using Postman for the requests, which may be a reason for the issue?

Could missing request headers be a cause for the issue?

I appreciate any insight anyone could provide!

**Comment by Alkaar** [comment, marked as answer]
Based on your error message @teddywagner, my assumption is that you didn't set the `Content-Type` header to `application/x-www-form-urlencoded`. What this does is escape special characters before your payload is sent over the web. Likely your `book_token` was not escaped so it was not received correctly on the other side which is why you are getting an `An invalid book token was provided.` error message.

**Reply by teddywagner** [reply]
Thank you that was the issue!

**Reply by pja66** [reply]
Hey @teddywagner ,
I am confused on how the book token is generated. Could you help me out with this? This is the last piece of the puzzle I need to understand

**Reply by pja66** [reply]
Thank you so much!

**Reply by bfiegoli** [reply]
Does anyone have a PR they may be able to put up with this change? Thanks so much!
